### DISCUSSION TITLE
Should we keep ch-ssh?

### DISCUSSION THREAD

**Discussion by reidpr** [hpc/charliecloud#1600] [Polls]
Charliecloud currently comes with a program called [`ch-ssh`](https://hpc.github.io/charliecloud/ch-ssh.html) that facilitates SSH connections that transparently start a container. It is bind-mounted into containers if (1) `/usr/bin/ch-ssh` exists in the image *and* (2) `--ch-ssh` is specified to `ch-run`.

We are considering removing `ch-ssh` (#1379) because:

1. We’re not aware that it gets used much.
2. The interface is awkward (`ch-run` arguments go in an environment variable that’s parsed with rules different from the shell).

This poll is to help us understand whether it’s getting used and we just don’t know it. Please especially respond if you *do* use and would be sad if it went away — if there aren’t any responses in this category, we’ll probably remove it in 0.33. Please also write a comment if you think that would be helpful.

The replacement would likely be a FAQ entry; you can do the same thing by adding `ch-run` to your `ssh` command line manually.

**Comment by qwofford** [comment]
Can you provide an example use-case?

**Reply by reidpr** [reply]
The intended use case was that you’re on Node 1 in a Charliecloud container, and you want to SSH to Node 2 and land in an equivalent Charliecloud container without manually invoking `ch-run`.

**Comment by reidpr** [comment]
At present we have one vote and it’s “no”. If nothing changes within the next week or so, we’ll remove the feature.

**Comment by reidpr** [comment]
OK I think we have our answer. `ch-ssh` will be gone in 0.33.
